### PR TITLE
Add static code analysis on commit and fix CodeQL errors

### PR DESCRIPTION
**Pull request by robertwu1**
Create an action to run static code analysis on demand.

After this is created, a separate PR will be created to fix linker issues described in #1506

### CODE REVIEW

**Comment by robertwu1** [comment]
Seems like there are issues in the autobuilder. https://github.com/github/codeql-action/issues/824

**Comment by ann0see** [comment]
What happens if you just use a normal build process instead of the auto-build provided by codeql?

**Comment by robertwu1** [comment]
> What happens if you just use a normal build process instead of the auto-build provided by codeql (probably you’re already doing that though)

Changing the build to something normal seems to work fine. Thanks! I will address the issues next week

**Comment by ann0see** [comment]
Ah, now similar issues show up in the CI: https://github.com/google/oboe/runs/5504278249

**Comment by ann0see** [comment]
Great! Thanks for fixing these errors

**Comment by ann0see** [comment]
Do you plan to add codeQl to your repo or just fix the overflow warnings? I think splitting the fixes in a separate PR would make sense such that you can decide if CodeQL should or shouldn’t be added.

**Comment by robertwu1** [comment]
Is there a downside of adding CodeQl to the CI besides the checks taking a bit longer?

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -34,4 +34,20 @@ jobs:
         chmod +x gradlew
         ./gradlew -q clean bundleDebug
         popd
- 
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
```
Comment (philburk) [inline .github/workflows/android.yml:39]: You might be able to combine the Initialize step with the build step above.

File: apps/OboeTester/app/src/main/cpp/analyzer/BaseSineAnalyzer.h [hunk 2]
```diff
@@ -145,8 +145,8 @@ class BaseSineAnalyzer : public LoopbackProcessor {
     bool transformSample(float sample, float referencePhase) {
         // Track incoming signal and slowly adjust magnitude to account
         // for drift in the DRC or AGC.
-        mSinAccumulator += sample * sinf(referencePhase);
-        mCosAccumulator += sample * cosf(referencePhase);
+        mSinAccumulator += static_cast<double>(sample) * static_cast<double>(sinf(referencePhase));
```
Comment (philburk) [inline apps/OboeTester/app/src/main/cpp/analyzer/BaseSineAnalyzer.h:148]: You probably only need to cast one operand. The other will get promoted automatically.

File: apps/OboeTester/app/src/main/cpp/analyzer/GlitchAnalyzer.h [hunk 3]
```diff
@@ -169,8 +169,10 @@ class GlitchAnalyzer : public BaseSineAnalyzer {
                 break;
 
             case STATE_WAITING_FOR_LOCK:
-                mSinAccumulator += sample * sinf(mInputPhase);
-                mCosAccumulator += sample * cosf(mInputPhase);
+                mSinAccumulator += static_cast<double>(sample)
```
Comment (philburk) [inline apps/OboeTester/app/src/main/cpp/analyzer/GlitchAnalyzer.h:172]: An alternative is to make mSinAccumulator a float.

File: apps/OboeTester/app/src/main/cpp/analyzer/LatencyAnalyzer.h [hunk 4]
```diff
@@ -73,8 +73,8 @@ static double calculateNormalizedCorrelation(const float *a,
 
     // Correlate a against b.
     for (int i = 0; i < windowSize; i++) {
-        float s1 = a[i];
-        float s2 = b[i];
+        double s1 = a[i];
```
Comment (philburk) [inline apps/OboeTester/app/src/main/cpp/analyzer/LatencyAnalyzer.h:76]: This code is already too slow.
Maybe better performance to make correlation, sumProducts, and sumSquares all float.

**Review by philburk — CHANGES_REQUESTED** [review]

File: apps/OboeTester/app/src/main/cpp/analyzer/GlitchAnalyzer.h (hunk 3 above)
Comment (robertwu1) [reply apps/OboeTester/app/src/main/cpp/analyzer/GlitchAnalyzer.h]: Leaving as is because it's used later in division

**Review by philburk — APPROVED** [review]

**Comment by robertwu1** [comment]
Fixes #1506

**Comment by ann0see** [comment]
Great thanks for fixing and merging!

**Pull request merged**